### ISSUE TITLE
Support for signup to sold-out event?

### ISSUE THREAD

**Issue by hermannsblum** [amiv-eth/amivapi#48]
One might want to signup for an event which is already full, in case somebody cancels his/her signup.

**Comment by hermannsblum** [comment]
I remember a discussion on this, have we changed anything in the api?

**Comment by NotSpecial** [comment]
IMHO we shoudl still accepts signups to full events.

There are two things we could do:
1. In the event, track how many people have signed up (or how many spots are left, but we have events with more signups then spots so i prefer the first solution)
2. for a signup, show which place you have in queue to give out some info (and mostly its FCFS anyway)

Opinions?

**Comment by fubu** [comment]
> 1. In the event, track how many people have signed up (or how many spots are left, but we have events with more signups then spots so i prefer the first solution)

I would include the number of signups - calculating the "spots left" is an easy task for any client. A new property (such as `signup_count`) is therefore required. (Right now the signup count could be inferred by looking at the length of the `Event.signups` list, but that list is probably only returned for event admins.)


This issue helped me remember another discussion about signups, so I opened issue #71

**Comment by NotSpecial** [comment]
I like that solution, I think it can be solved easily with a hook to event requests which will add the property

**Comment by fubu** [comment]
I have not yet worked with Eve that much, but can't you add the field `signup_count` on the schema for `Events` and implement it on the model?

```python
class Event(Base):
    # ...

    @property
    def signup_count(self):
        return len(self.signups)
```

**Comment by NotSpecial** [comment]
Well, definetly worth a shot - I'll report back if it works that way

**Comment by Leonidaz0r** [comment]
signup_count was implemented in 88fcfc76e5b603a6d9424c0ebed22fbdccd9ef51

**Comment by Tar-Tarus** [comment]
The signup_count was implemented as commented above and the queue *position* field has been added as of b3400c5841bb889304682ea7cd0a07e24e5f00ad
This issue can therefore, in my opinion be closed as the necessary enhancements have been implemented and the core discussion is continuing in other issue threads.
